caffe2/operators/utility_ops.cc

```diff
@@ -14,6 +14,31 @@ bool WeightedSumGradientOp<CPUContext>::RunOnDevice() {
   return DoRunWithType<float>();
 }
 
+std::vector<TensorShape> WeightedSumShapeInference(
+    const OperatorDef& def,
+    const vector<TensorShape>& in) {
+  vector<TensorShape> out(1);
+  out[0] = in[0];
+  return out;
+}
+
+OpSchema::Cost CostInferenceForWeightedSum(
+    const OperatorDef& def,
+    const vector<TensorShape>& in) {
+  CAFFE_ENFORCE_EQ(
+      in.size() % 2, 0, "WeightedSum requires an even number of inputs");
+  struct OpSchema::Cost c;
+
+  const auto& X0 = in[0];
+  const auto& nElem = nElemFromDim(X0);
+  const auto& nInputs = in.size();
+  c.flops = (nInputs - 1) * nElem;
```

Comment:
We also have multiplications.
``
c.flops = (2 * nInputs - 1) * nElem;
``

Reply:
The count is correct as is.  The formulation should really be `(2 * nInputs / 2 - 1) * nElem`.  The `nInputs / 2` accounts for the number of pairs.